비디오 리스트 관련 기능 추가

hyokualexkwon opened the pull request:
## server

- 영상 리스트가 popular 순으로 정렬될 때, period query를 사용하여 week, month, year, all 단위로 필터링 할 수 있습니다.
- cors 옵션을 활성화 했습니다. whiltelist origin 항목은 .env의 CORS_ALLOWED_ORIGINS 항목으로 관리합니다.

## client

- 핫리스트 페이지에서 api를 활용하여 데이터를 불러옵니다. 이 때, 간결한 interface와 caching 기능을 제공하는 react-query 모듈을 사용했습니다.
- 기간 필터링을 누를 때마다 새로운 쿼리 내용으로 요청을 생성합니다.
- 최신 영상 페이지에서 api를 활용하여 데이터를 불러옵니다.

hyokualexkwon committed:
[client] | refactor(AppBar): 앱 바 검색 바 가운데 정렬

hyokualexkwon committed:
[client] | refactor(AppBar): 앱바 container 최대 넓이 삭제

drawer가 없는 상태에서 appbar에 최대넓이를 적용하던 속성을 삭제했습니다. appbar에는 최대 넓이가 적용되지 않는 것이 ui 더 자연스러워집니다

hyokualexkwon committed:
[client] | refactor(SearchBar): 검색 창 스타일 수정

hyokualexkwon committed:
[client] | refactor(Filter): Filter 인터페이스 변경

필터 옵션을 value와 label속성을 담은 배열로 전달 받습니다.

hyokualexkwon committed:
[client] | feature(Hotlist): useHotlistVideos hook 작성

react-query 모듈을 설치하고 Hotlist에서 사용할 영상 리스트를 조회하는 query hook을 작성했습니다.

hyokualexkwon committed:
[client] | feature(Hotlist): 핫리스트 영상 조회 및 데이터 렌더링

useHotlistVideos hook을 활용하여 영상 리스트를 조회하고, 렌더링합니다.

hyokualexkwon committed:
[server] | feat(videos): 영상 리스트 조회 시 기간 필터 작성

/videos api에 대해, sort 옵션이 popular인 경우 일주일, 한달, 일년, 또는 전체 기간에 대해 필터를 추가할 수 있는 옵션을 작성했습니다.

hyokualexkwon committed:
[server] | feat(cors): cors옵션 설정 및 허용 origin을 .env에 추가

hyokualexkwon committed:
[database] | refactor(comment, video): updatePopularity에 BeforeUpdate추가

hyokualexkwon committed:
[client] | feat(VideoItem): 영상 시간 parser 작성

영상 재생 시간을 간결하게 변환하는 parser를 작성하고, 영상 썸네일 위에 렌더링했습니다.

hyokualexkwon committed:
[client] | feat(VideoItem): timeago 모듈을 활용하여 시간 표현

timeago 모듈을 설치하고, 한국어로 영상 업로드 시간을 표현하도록 VideoItem 컴포넌트를 작성했습니다.

hyokualexkwon committed:
[client] | feat(): react-fetching-library 모듈 설정

앱에서 사용할 데이터 통신 모듈인 react-fetching-library를 설치하고, 본 모듈의 caching 기능을 사용할 수 있도록 설정을 작성했습니다.

hyokualexkwon committed:
[client] | 무한 스크롤 페이지 네이션 작성

- react-infinite-scroll-component 모듈을 사용하여 무한 스크롤 페이지네이션을 작성했습니다.
- useHotlistVideos hook을 작성하여 핫리스트 영상 목록을 fetching 하는 로직을 담았습니다. 페이지가 바뀌거나 기간 필터의 내용이 변경될 때, 새로운 데이터를 반환합니다.

hyokualexkwon committed:
[client] | feat(Latest): 최신 영상 페이지 비디오 리스트 작성

Hotlist와 같은 구현방법으로 Latest 페이지를 작성했습니다. 추후에 공통으로 사용하는 컴포넌트를 만들어서 양 페이지에서 사용할 예정입니다.

KimJye commented on packages/client/components/CircularProgress/index.tsx:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
이 컴포넌트는 무엇에대한 컴포넌트인가요??

hyokualexkwon replied:
로딩중일때 빙글빙글 돌아가는 UI입니다~
https://material-ui.com/api/circular-progress/

KimJye commented on packages/client/components/AppBar/index.tsx:
```diff
@@ -21,7 +21,7 @@ export const AppBar: React.FunctionComponent<AppBarProps> = ({ maxWidth }) => {
   return (
     <S.AppBar>
-      <S.Container maxWidth={maxWidth}>
+      <S.Container>
```
maxwidth를 왜 지우셧나요??

hyokualexkwon replied:
drawer가 없는 상태에서 appbar에 maxwidth를 적용하는 ui를 없앴습니다.~

KimJye commented on packages/client/components/SearchBar/index.tsx:
```diff
@@ -32,7 +32,7 @@ export const SearchBar: React.FunctionComponent<SearchBarProps> = ({
 
       <S.Input isActive={isActive}>
-        <SearchSVG width={20} height={20} viewBox={'0, 0, 20, 20'} />
+        <SearchSVG width={20} height={20} viewBox="0 0 24 24" />
```
저번 PR때 width={20} height={20}와 같이 코드의 통일성을 위해서 viewBox={"0 0 20 20"}로 했으면 좋겠다라는 리뷰를 받아서 viewBox={"0 0 20 20"}로 했었는데, 위와같이 변경하신 이유가 있을까요??

hyokualexkwon replied:
실제 html DOM에는 viewBox="0 0 24 24" 로 표현되거든요. 네이티브 인터페이스와 더 가까우면 좋을 것 같습니다.

KimJye commented on packages/server/src/video/video.controller.ts:
```diff
@@ -13,10 +13,11 @@ export class VideoController {
   public async getVideos(
     @Query() videosQueryDto: VideosQueryDto,
   ): Promise<Video[]> {
```
제가 생각했을때, 이 요청에대한 응답으로 필요한 video의 데이터만 주는것은 어떨까요?
여기서 응답으로 필요한 video의 데이터들은 id, title, views, thumbnail, playtime, createdAt또는 updatedAt, userId 라고 생각합니당

hyokualexkwon replied:
좋은 의견이네요 반영하겠습니다!

hyokualexkwon committed:
[server] | refactor(VideoResponseDto): video 컨트롤러의 반환 형태 재정의

Video controller의 getVideos가 반환하는 데이터 형태가 필요한 것만 포함하도록 작성했습니다. 이때, 필요한 데이터는 VideoResponseDto에 정의합니다.

KimJye reviewed (approved):
고생많으셧씁니다

Atercatus commented on packages/client/components/AppBar/index.tsx:
```diff
@@ -40,7 +40,7 @@ export const AppBar: React.FunctionComponent<AppBarProps> = ({ maxWidth }) => {
 
           <button onClick={handleSearchBar}>
-            <SearchSVG width={23} height={24} />
+            <SearchSVG width={24} height={24} />
```
변수로 바꾸고 싶어요

Atercatus commented on packages/client/components/Filters/index.tsx:
```diff
@@ -3,30 +3,30 @@ import React from 'react';
-  onClick: (filter: string) => void;
+  filters: Array<{ label: string; value: string }>;
+  activeValue: string;
+  onClick: (value: string) => void;
```
어떤 value인가요?

Atercatus commented on packages/client/libs/fetching/cache.ts:
```diff
@@ -0,0 +1,11 @@
+import { createCache, QueryResponse } from 'react-fetching-library';
+import { Cache } from 'react-fetching-library/lib/cache/cache.types';
+
+export const cache: Cache<QueryResponse> = createCache(
```
어떤 역할을 하는 함수인가요?

Atercatus commented on packages/client/libs/parsePlaytime.ts:
```diff
@@ -0,0 +1,37 @@
+    return '';
+  }
+
+  return `${hours}:`;
```
```
return hours ? `${hours}` : ''; 
```
는 어떤가요?

Atercatus commented on packages/client/libs/parsePlaytime.ts:
```diff
@@ -0,0 +1,37 @@
+};
+
+const parseMinutes = (minutes: number, hours: number) => {
+  if (minutes < 10 && hours !== 0) {
```
따로 함수로 빼서 함수명을 통해 읽기 좋게 하면 더 좋을 거 같아요

Atercatus commented on packages/client/pages/_app.tsx:
```diff
@@ -1,16 +1,42 @@
 import React from 'react';
 import App from 'next/app';
 import { ThemeProvider } from 'styled-components';
+import 'isomorphic-unfetch';
```
어떤역할을 수행하나요?

Atercatus commented on packages/server/src/common/pipes/get-videos.pipe/index.ts:
```diff
@@ -25,20 +21,49 @@ export class GetVideosPipe implements PipeTransform {
+    return sort && (sort === LATEST || sort === POPULAR);
+  }
+
+  private validatePeriod(period: string, sort: string) {
```
함수로 묶어서 더 명확하게 표현하면 좋을 거 같습니다.

hyokualexkwon committed:
Merge branch 'develop' into feature/video_list_20191123
